packages/server/lib/controllers/proxy.controller.ts

```diff
@@ -294,13 +294,31 @@ See https://docs.nango.dev/guides/proxy#proxy-requests for more information.`
      * @param {AxiosError} error
      * @param {attemptNumber} number
      */
-    private retry = async (activityLogId: number, error: AxiosError, attemptNumber: number): Promise<boolean> => {
+    private retry = async (activityLogId: number, config: ProxyBodyConfiguration, error: AxiosError, attemptNumber: number): Promise<boolean> => {
+        if (config.template.proxy && config.template.proxy.retry_header) {
+            const retryHeader = config.template.proxy.retry_header;
+            const retryHeaderVal = error?.response?.headers[retryHeader] || error?.response?.headers[retryHeader.toLowerCase()];
+
+            if (retryHeaderVal) {
+                const retryAfter = Number(retryHeaderVal);
+                const content = `Retry header was parsed successfully, retrying after ${retryAfter} seconds`;
+
+                await createActivityLogMessage({
+                    level: 'error',
+                    activity_log_id: activityLogId,
+                    timestamp: Date.now(),
+                    content
+                });
+
+                await new Promise((resolve) => setTimeout(resolve, retryAfter * 1000));
+
+                return true;
+            }
+        }
         if (
             error?.response?.status.toString().startsWith('5') ||
             error?.response?.status === 429 ||
```

Comment:
Unfortunately, GitHub returns rate-limit errors with `403` ([docs](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit)) 😕 

I don't know what'd be the bullet-proof solution here—maybe adding a list of additional status codes to retry? What do you think? 🤔

Reply:
Yeah 😦, will just add 403 as a possible response code to retry on as well here.